src/libertem/viz/base.py

```diff
@@ -0,0 +1,218 @@
+        dict of kwargs passed to Pillow when saving the image, can be used to set
+        the file format, quality, ...
+
+    Returns
+    -------
+
+    BytesIO
+        a buffer containing the result image (as PNG/JPG/... depending on save_kwargs)
+    """
+    if save_kwargs is None:
+        save_kwargs = {'format': 'png'}
+    # see also: https://stackoverflow.com/a/10967471/540644
+    im = Image.fromarray(result)
+    buf = BytesIO()
+    im = im.convert(mode="RGB")
+    im.save(buf, **save_kwargs)
+    buf.seek(0)
+    return buf
+
+
+def visualize_simple(result, colormap=None, logarithmic=False, vmin=None, vmax=None, damage=None):
+    """
+    Normalize and visualize ``result`` with ``colormap`` and return the
+    resulting RGBA data as an array.
+
+    Parameters
+    ----------
+    result : numpy.ndarray
+        2d array of intensity values
+
+    colormap : matplotlib colormap or None
+        colormap used for visualizing intensity values, defaults to ColormapCubehelix()
+
+    Returns
+    -------
+
+    np.array
+        A numpy array of shape (Y, X, 4) containing RGBA data, suitable for
+        passing to `Image.fromarray` in PIL.
+    """
+    if logarithmic:
+        # Convert to the smallest dtype that supports subtractions
+        dtype = np.result_type(result, np.int8)
+        result = result.astype(dtype)
+        cnorm = colors.LogNorm
+        result = result - np.min(result) + 1
+    else:
+        cnorm = colors.Normalize
+    if colormap is None:
+        colormap = cm.gist_earth
+    norm = _get_norm(result, norm_cls=cnorm, vmin=vmin, vmax=vmax, damage=damage)
+    shape = result.shape
+    normalized = norm(result.reshape((-1,))).reshape(shape)
+    colored = colormap(normalized, bytes=True)
+    return colored
+
+
+def get_plottable_2D_channels(buffers):
+    return [
+        k
+        for k, buf in buffers.items()
+        # 2D data, removing axes of size 1
+        if len(buf.data.squeeze().shape) == 2
+    ]
+
+
+class Live2DPlot:
+    """
+    Base plotting class for interactive use. Please see the subclasses for concrete details.
+    """
+    def __init__(
+            self, dataset, udf, roi=None, channel=None, udfresult=None
+    ):
+        """
+        Construct a new `LivePlot`
+
+        Parameters
+        ----------
+        dataset : DataSet
+            The dataset on which the UDf will be run. This allows to determine the
+            shape of the plots for initialization.
+
+        udf : UDF
+            The UDF instance this plot is associated to. This needs to be
+            the same instance that is passed to :meth:`~libertem.api.Context.run_udf`.
+
+        roi : numpy.ndarray or None
+            Region of interest (ROI) that the UDF will be run on. This is necessary for UDFs
+            where the `extra_shape` parameter of result buffers is a function of the ROI,
+            such as :class:`~libertem.udf.raw.PickUDF`.
+
+        channel : str or function udf_result -> ndarray
```

Comment:
`channel` is now a bit more complicated - docstring needs an update. Channel can also be `None`, in which case the first channel will be plotted.

I was thinking if it's maybe possible to keep the simple "postprocessing"-style for the easy/common case where you just want to throw a function on the result before visualization. Something I just came up with:

```python
# easiest way: no channel at all, first channel is plotted
Live2DPlot(dataset=ds, udf=udf, channel=None)

# channel as str
Live2DPlot(dataset=ds, udf=udf, channel="intensity")

# channel w/ postprocessing fn
Live2DPlot(dataset=ds, udf=udf, channel=("intensity", np.log1p))

# complete control, including combinations of channels, damage etc.
Live2DPlot(dataset=ds, udf=udf, channel=my_extraction_function)
```

Thoughts?

Reply:
Oh, this looks great! Will complicate things abit behind the scenes, but should be manageable for users and would allow passing functions through the `plots=...` parameter in the `Context` methods. I'll work on that next.